src/puppetlabs/kitchensink/file.clj

```diff
@@ -0,0 +1,84 @@
+(ns puppetlabs.kitchensink.file
+  (:import [java.io BufferedWriter FileOutputStream OutputStreamWriter]
+           [java.nio.file CopyOption Files LinkOption Paths StandardCopyOption]
+           [java.nio.file.attribute FileAttribute PosixFilePermissions]))
+
+(defn str->path
+  [path]
+  (Paths/get path (into-array String [])))
+
+(defn get-perms
+  "Returns the currently set permissions of the given file path."
+  [path]
+  (-> (str->path path)
```

Comment:
any particular reason why you didn't do
```
(-> path
    (str->path)
    (...))
```
?

Reply:
I just pulled this from puppetserver, but I do prefer the more compact form if there's a trivial conversion at the start.